src/retasc/run.py

```diff
@@ -0,0 +1,111 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import logging
+import os
+import re
+from collections import defaultdict
+from collections.abc import Iterator
+
+from retasc.jira_client import DryRunJiraClient, JiraClient
+from retasc.models.parse_rules import parse_rules
+from retasc.models.release_rule_state import ReleaseRuleState
+from retasc.models.rule import Rule
+from retasc.product_pages_api import ProductPagesApi
+from retasc.report import Report
+from retasc.requests_session import requests_session
+from retasc.runtime_context import RuntimeContext
+from retasc.templates.template_manager import TemplateManager
+
+logger = logging.getLogger(__name__)
+
+RE_VERSION = re.compile(r"^\w+-(?P<major>\d+)(?:[-.](?P<minor>\d+))?")
```

Comment:
There could be beta as well.

Reply:
https://pp.engineering.redhat.com/rhel-10.0.beta/platforms/ was marked as unsupported a few days back. 
So think the next beta for rhel 11 would be a few years later. So probably  keeps the logic simple for now as what you have.